pkg/daemon/ceph/osd/daemon.go

```diff
@@ -84,10 +86,42 @@ func StartOSD(context *clusterd.Context, osdType, osdID, osdUUID, lvPath string,
 		if err := releaseLVMDevice(context, volumeGroupName); err != nil {
 			return fmt.Errorf("failed to release device from lvm. %+v", err)
 		}
+
 	}
+
 	return nil
 }
 
+func handleTerminate(context *clusterd.Context, lvPath, volumeGroupName string) error {
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, syscall.SIGTERM)
+	for {
+		select {
+		case <-sigc:
+			logger.Infof("shutdown signal received, exiting...")
+
+			pid, err := context.Executor.ExecuteCommandWithOutput(false, "", "fuser", "-a", lvPath)
+			if err != nil {
+				return fmt.Errorf("failed to retrieve process ID for - %s. Error %+v", lvPath, err)
+			}
+
+			logger.Infof("process ID for logical volume  %s : %s", lvPath, pid)
+			// shut down the osd-ceph process so that lvm release does not show device in use error.
+			if pid != "" {
+				if err := context.Executor.ExecuteCommand(false, "", "kill", pid); err != nil {
```

Comment:
Since the mounts have been added, I'd suggest you test without other changes like killing the osd. Perhaps the bind mounts were all we were missing. I'm hoping calling `releaseLVMDevice()` after the ceph-osd process exits naturally (line 86) will be sufficient now.